### PR TITLE
methods "getStatistic", "readSourceFile", "writeResultToFile" implemented

### PR DESCRIPTION
**Pull request by lavrivskyi**

### CODE REVIEW

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 1]
```diff
@@ -1,8 +1,59 @@
...
+    private static final String SUPPLY = "supply";
+    private static final String BUY = "buy";
+    private static final String RESULT = "result";
 
     public void getStatistic(String fromFileName, String toFileName) {
+        String resultString = readSourceFile(fromFileName);
+        writeResultToFile(resultString, toFileName);
+    }
+
+    public String readSourceFile(String fromFileName) {
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/WorkWithFile.java:20]: lets make this method private

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 2]
```diff
@@ -1,8 +1,59 @@
...
+                    .append(System.lineSeparator())
+                    .append(RESULT + ",").append(supplyAmount - buyAmount)
+                    .append(System.lineSeparator());
+            return resultBuilder.toString();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't read file", e);
+        }
+    }
 
+    public void writeResultToFile(String lineToWrite, String toFileName) {
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/WorkWithFile.java:52]: lets make this method private too

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 3]
```diff
@@ -1,8 +1,59 @@
...
     public void getStatistic(String fromFileName, String toFileName) {
+        String resultString = readSourceFile(fromFileName);
+        writeResultToFile(resultString, toFileName);
+    }
+
+    public String readSourceFile(String fromFileName) {
+        try {
+            int supplyAmount = 0;
+            int buyAmount = 0;
+            BufferedReader reader = new BufferedReader(new FileReader(fromFileName));
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/WorkWithFile.java:24]: lets use try-with-resources

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 4]
```diff
@@ -1,8 +1,59 @@
...
+            String line = reader.readLine();
+            String[] tempArray;
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/WorkWithFile.java:26]: rowData

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 5]
```diff
@@ -1,8 +1,59 @@
...
+
+            resultBuilder.append(SUPPLY + ",").append(supplyAmount)
+                    .append(System.lineSeparator())
+                    .append(BUY + ",").append(buyAmount)
+                    .append(System.lineSeparator())
+                    .append(RESULT + ",").append(supplyAmount - buyAmount)
+                    .append(System.lineSeparator());
+            return resultBuilder.toString();
+        } catch (Exception e) {
+            throw new RuntimeException("Can't read file", e);
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/WorkWithFile.java:48]:
```suggestion
            throw new RuntimeException("Can't read file " + fromFileName, e);
```

lets make our exceptions more informative

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 6]
```diff
@@ -1,8 +1,59 @@
...
+        }
+    }
 
+    public void writeResultToFile(String lineToWrite, String toFileName) {
+        try (BufferedWriter bufferedWriter = new BufferedWriter(new FileWriter(toFileName, true))) {
+            bufferedWriter.write(lineToWrite);
+        } catch (Exception e) {
+            throw new RuntimeException("Can't write to file", e);
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/WorkWithFile.java:56]:
```suggestion
            throw new RuntimeException("Can't write to file " + toFileName, e);
```

**Review by MaliukDaria — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 6 above)
Comment (lavrivskyi) [reply src/main/java/core/basesyntax/WorkWithFile.java]: fixed

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 5 above)
Comment (lavrivskyi) [reply src/main/java/core/basesyntax/WorkWithFile.java]: fixed

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 4 above)
Comment (lavrivskyi) [reply src/main/java/core/basesyntax/WorkWithFile.java]: fixed

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 2 above)
Comment (lavrivskyi) [reply src/main/java/core/basesyntax/WorkWithFile.java]: fixed

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 3 above)
Comment (lavrivskyi) [reply src/main/java/core/basesyntax/WorkWithFile.java]: fixed

File: src/main/java/core/basesyntax/WorkWithFile.java [hunk 7]
```diff
@@ -1,8 +1,58 @@
...
+    }
+
+    private String readSourceFile(String fromFileName) {
+        try (BufferedReader reader = new BufferedReader(new FileReader(fromFileName))) {
+            int supplyAmount = 0;
+            int buyAmount = 0;
+            String row = reader.readLine();
+            String[] rowData;
+            StringBuilder resultBuilder = new StringBuilder();
+
```
Comment (MaliukDaria) [inline src/main/java/core/basesyntax/WorkWithFile.java:27]: давай уберем ненужные пустые строки из кода

**Review by MaliukDaria — CHANGES_REQUESTED** [review]

File: src/main/java/core/basesyntax/WorkWithFile.java (hunk 7 above)
Comment (lavrivskyi) [reply src/main/java/core/basesyntax/WorkWithFile.java]: fixed

**Review by MaliukDaria — APPROVED** [review]

**Pull request closed without merging**